lib/beeline/topology.ex

```diff
@@ -3,14 +3,13 @@ defmodule Beeline.Topology do
 
   @behaviour GenServer
 
+  import Beeline.ProcessNaming, only: [name: 2]
   alias __MODULE__.StageSupervisor
 
   defstruct [:supervisor_pid, :config]
 
   def start_link(config) do
```

Comment:
If this were `start_link(config, options \\ [])` you could just pass options directly to GenServer.  Then, from another higher level module in your API, write the layer which handles the process name shenanigans.  I feel a soapbox rant coming on.